src/rules/noDuplicateImportsRule.ts

```diff
@@ -41,21 +47,43 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        return this.applyWithFunction(sourceFile, (ctx: Lint.WalkContext<void>) =>
```

Comment:
This looks like it works, but to simplify a bit, you can pass an options object as a third argument to `applyWithFunction`. That'll allow you to access it on `ctx.options` instead of passing a new parameter.